tests/suites/storage/charm_storage.sh

```diff
@@ -0,0 +1,146 @@
+	juju create-storage-pool -m "${model_name}" tempy tmpfs size=1G
+	juju create-storage-pool -m "${model_name}" ebsy ebs size=1G
+	echo "create-storage-pool PASSED"
+
+	# Assess the above created storage pools.
+	echo "Assessing storage pool"
+	if [ "${BOOTSTRAP_PROVIDER:-}" == "ec2" ]; then
+		juju list-storage-pools -m "${model_name}" --format json | jq '.ebs | .provider' | check "ebs"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.["ebs-ssd"] | .provider' | check "ebs"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.tmpfs | .provider' | check "tmpfs"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.loop | .provider' | check "loop"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.rootfs | .provider' | check "rootfs"
+	else
+		juju list-storage-pools -m "${model_name}" --format json | jq '.rooty | .provider' | check "rootfs"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.tempy | .provider' | check "tmpfs"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.loopy | .provider' | check "loop"
+		juju list-storage-pools -m "${model_name}" --format json | jq '.ebsy | .provider' | check "ebs"
+	fi
+	echo "Storage pool PASSED"
+
+	# Assess charm storage with the filesystem storage provider
+	echo "Assessing filesystem rootfs"
+	juju deploy ./testcharms/charms/dummy-storage-fs --series="jammy" --storage data=rootfs,1G
+	wait_for "dummy-storage-fs" ".applications"
+	# assert the storage kind name
+	juju list-storage --format json | jq '.storage[] | .kind' | check "filesystem"
+	# assert the storage label name
+	juju list-storage --format json | jq '.storage | keys | .[]' | check "data/0"
+	# assert the unit attachment name
+	juju list-storage --format json | jq '.storage[] | .attachments | .units | keys | .[]' | check "dummy-storage-fs/0"
+	# assert the attached unit state
+	juju list-storage --format json | jq '.storage[] | .attachments | .units[] | .life' | check "alive"
+	echo "Filesystem rootfs PASSED"
+	# remove the application
+	juju remove-application dummy-storage-fs
+	wait_for "{}" ".applications"
+
+	# Assess charm storage with the filesystem storage provider
+
+	echo "Assessing block loop disk 1"
+	juju deploy ./testcharms/charms/dummy-storage-lp --series="jammy" --storage disks=loop,1G
+	wait_for "dummy-storage-lp" ".applications"
+	# assert the storage kind name
+	juju list-storage --format json | jq '.storage | .["disks/1"] | .kind' | check "block"
+	# assert the storage label name
+	juju list-storage --format json | jq '.storage| keys[0]' | check "disks/1"
+	# assert the unit attachment name
+	juju list-storage --format json | jq '.storage |.["disks/1"]|.attachments |.units | keys | .[]' | check "dummy-storage-lp/0"
+	# assert the attached unit state
+	juju list-storage --format json | jq '.storage[] | .attachments |.units[] | .life' | check "alive"
+	echo "Block loop disk 1 PASSED"
```

Comment:
Happy to add that as well.

Reply:
I figured from the docs the reason as to why we shouldnot assert size is because it grows. the allocated 1G size is the minimum if the charm itself does not declare it in the storage definition.

Reply:
Partially addressed this above. A rootfs storage unit (I'm pretty sure) cannot be any size other than the capacity of the machines filesystem storage, hence why 7.5Gb is so much larger than 1Gb asked for

A check such as `if [ $requested_storage -le $acquired_storage ] ; then exit 1 ; fi` would do for this test